rasa/nlu/featurizers/sparse_featurizer/text_featurizer.py

```diff
@@ -0,0 +1,313 @@
+        "digit": lambda word: word.text.isdigit(),
+    }
+
+    def __init__(
+        self,
+        component_config: Dict[Text, Any],
+        feature_to_idx_dict: Optional[Dict[Text, Any]] = None,
+    ):
+        super().__init__(component_config)
+
+        if feature_to_idx_dict is None:
+            self.feature_to_idx_dict = {}
+        else:
+            self.feature_to_idx_dict = feature_to_idx_dict
+
+        self._check_pos_features_and_spacy()
+
+    def _check_pos_features_and_spacy(self):
+        import itertools
+
+        features = set(
+            itertools.chain.from_iterable(self.component_config.get("features", []))
+        )
+        self.pos_features = "pos" in features or "pos2" in features
+
+        if self.pos_features and spacy is None:
+            raise ImportError(
+                "Failed to import `spaCy`. `spaCy` is required for POS features. "
+                "See https://spacy.io/usage/ for installation instructions."
+            )
+
+    def train(
+        self,
+        training_data: TrainingData,
+        config: Optional[RasaNLUModelConfig] = None,
+        **kwargs: Any,
+    ) -> None:
+        self.feature_to_idx_dict = self._create_feature_to_idx_dict(training_data)
+
+        for example in training_data.training_examples:
+            self._create_text_features(example)
+
+    def process(self, message: Message, **kwargs: Any) -> None:
+        self._create_text_features(message)
+
+    def _create_text_features(self, message: Message) -> None:
+        """Convert incoming messages into sparse features using the configured
+        features."""
+
+        words = self._convert_to_words(message)
+        word_features = self._words_to_features(words)
+        features = self._features_to_one_hot(word_features)
+        features = self._combine_with_existing_sparse_features(
+            message, features, feature_name=SPARSE_FEATURE_NAMES[TEXT_ATTRIBUTE]
+        )
+        message.set(SPARSE_FEATURE_NAMES[TEXT_ATTRIBUTE], features)
+
+    def _features_to_one_hot(
+        self, word_features: List[Dict[Text, Any]]
+    ) -> scipy.sparse.spmatrix:
+        """Convert the word features into a one-hot presentation using the indices
+        in the feature-to-idx dictionary."""
+
+        vec = self._initialize_feature_vector(len(word_features))
+
+        for word_idx, word_features in enumerate(word_features):
+            for feature_key, feature_value in word_features.items():
+                if (
+                    feature_key in self.feature_to_idx_dict
+                    and str(feature_value) in self.feature_to_idx_dict[feature_key]
+                ):
+                    feature_idx = self.feature_to_idx_dict[feature_key][
+                        str(feature_value)
+                    ]
+                    vec[word_idx][feature_idx] = 1
+
+        # set vector of CLS token to sum of everything
+        vec[-1] = np.sum(vec, axis=0)
```

Comment:
Do you think `sum` is the correct metric here? In case of sum the vector of the CLS token will not contain any zeros.

Reply:
what do you mean?

Reply:
Just wanted to check if you agree that we should take the sum for the CLS token vector. But I guess as you did not comment, it is fine :)

Reply:
yeah, I think it's fine, like bag-of-words